bot/exts/recruitment/talentpool/_cog.py

```diff
@@ -407,6 +407,78 @@ async def end_nomination_command(self, ctx: Context, user: MemberOrUser, *, reas
+
+    @nomination_append_group.command(name="reason")
+    @has_any_role(*STAFF_ROLES)
+    async def append_reason_command(
+        self,
+        ctx: Context,
+        nominee_or_nomination_id: UnambiguousMemberOrUser | int,
+        nominator: UnambiguousMemberOrUser | None = None,
+        *,
+        reason: str
+    ) -> None:
+        """
+        Append additional text to the nomination reason of a specific nominator for a given nomination.
+
+        If nominee_or_nomination_id resolves to a member or user,
+        append the text to the currently active nomination for that person.
+        Otherwise, if it's an int, look up that nomination ID to edit.
+
+        If no nominator is specified, assume the invoker is editing their own nomination reason.
+        Otherwise, edit the reason from that specific nominator.
+
+        Raise a permission error if a non-mod staff member invokes this command on a
+        specific nomination ID, or with an nominator other than themselves.
+        """
+        # If not specified, assume the invoker is editing their own nomination reason.
+        nominator = nominator or ctx.author
+
+        if not any(role.id in MODERATION_ROLES for role in ctx.author.roles):
+            if ctx.channel.id != Channels.nominations:
+                await ctx.send(f":x: Nomination edits must be run in the <#{Channels.nominations}> channel.")
+                return
+
+            if nominator != ctx.author or isinstance(nominee_or_nomination_id, int):
+                raise BadArgument(
+                    "Only moderators can edit specific nomination IDs, "
+                    "or the reason of a nominator other than themselves."
+                )
+
+        if isinstance(nominee_or_nomination_id, int):
+            nomination_id = nominee_or_nomination_id
+            original_nomination = await self.api.get_nomination(nomination_id=nomination_id)
+            if original_nomination:
+                nomination_entries = original_nomination.entries
+            else:
+                await ctx.send(f":x: {nominee_or_nomination_id} is not a valid nomination id.")
+                return
+        else:
+            active_nominations = await self.api.get_nominations(user_id=nominee_or_nomination_id.id, active=True)
+            if active_nominations:
+                nomination_entries = active_nominations[0].entries
+            else:
+                await ctx.send(f":x: {nominee_or_nomination_id.mention} doesn't have an active nomination.")
+                return
+
+        old_reason = next (e.reason for e in nomination_entries if e.actor_id == nominator.id)
+
+        if old_reason is not None and reason is not None:
+            add_period = not old_reason.endswith((".", "!", "?"))
+            reason = old_reason + (". " if add_period else " ") + reason
+
+        await self._edit_nomination_reason(
+            ctx,
+            target=nominee_or_nomination_id,
```

Comment:
what happens here if the nominator has not previously nominated them? This would be the case when old_reason is  None.

Reply:
The `edit_nomination_reason` function can handle the case when the nominator doesn't already have a nomination. From that function: 
```
        try:
            nomination = await self.api.edit_nomination_entry(nomination_id, actor_id=actor.id, reason=reason)
        except ResponseCodeError as e:
            match (e.status, e.response_json):
                case (400, {"actor": _}):
                    await ctx.send(f":x: {actor.mention} doesn't have an entry in this nomination.")
                    return
```
Currently for `!tp edit reason` this is where that case is handled as well.

Reply:
Looks good then, thanks!